lib/rubygems/test_case.rb

```diff
@@ -308,6 +308,14 @@ def setup
     Dir.chdir @tempdir
 
     ENV['HOME'] = @userhome
+    FileUtils.mkdir_p File.join(@userhome, ".gem")
+    File.open(File.join(@userhome, ".gemrc"), "w"){|f| f.puts "--- {}" }
+    temp_cred = File.join(@userhome, '.gem', 'credentials')
+    FileUtils.mkdir_p File.dirname(temp_cred)
+    File.open temp_cred, 'w', 0600 do |fp|
+      fp.puts ':rubygems_api_key: 701229f217cdf23b1344c7b4b54ca97'
+    end
+    ENV['XDG_DATA_HOME'] = File.join @userhome, ".local", "share"
```

Comment:
This is the default fallback, do we need to set this environment variable at all?